delegate-examples/shared/src/main/scala/zio/macros/delegate/EnrichWithExample.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2017-2019 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.macros.delegate
+
+import zio._
+import zio.clock.Clock
+import zio.blocking.Blocking
+import zio.console.Console
+
+object EnrichWithExample {
+
+  val blockingWithDeps: ZIO[Console, Throwable, Blocking] = ZIO.succeed(Blocking.Live)
+
+  val enrichClock        = enrichWith[Clock](Clock.Live)
```

Comment:
Can `R` be inferred? I guess it would be as `Clock.Live.type`, but that should be perfectly ok and "fit" the `Clock` requirement.

Reply:
The problem is that Mix needs a trait to mix in and will not automatically try to widen by result.
So here we basically declare what we want to implement / provide